### DISCUSSION TITLE
Add nomad backend

### DISCUSSION THREAD

**Discussion by 6543** [woodpecker-ci/woodpecker#2158] [Request a Feature]
https://www.nomadproject.io/

**Comment by 6543** [comment]
it was mentioned in matrix dev group ... https://matrix.to/#/!ElrAiHCaPpSDVVcayj:obermui.de/$qNE3aC7H4CtTz26A-9aC55FRSFwSYV5pK91ZV1jwY3U?via=obermui.de&via=t2bot.io&via=matrix.org

**Comment by jinnatar** [comment]
Since the draft PR for this (https://github.com/woodpecker-ci/woodpecker/pull/1697) seems to question the need for a Nomad agent backend, Ill add a few notes re worthiness of inclusion:
- There was an implication Nomad was just a special kind of k8s. Nomad is *not* built on top of k8s and is a direct competitor. It's a standalone orchestrator capable of running workloads in a myriad of ways, including containers on Docker/ Podman etc.
- Obviously, it's not nearly as popular as k8s and is no longer Open Source. Both are minuses against inclusion, just mentioning for the sake of completeness. I still use it because it's significantly easier than k8s in my personal opinion.
- This is still the one feature I miss from Drone (https://docs.drone.io/runner/nomad/installation/). While I can run the woodpecker-agent with the default Docker backend on Nomad just fine, I lose visibility since the workflows are just injected into the "implementation details" of specific nodes where the agents are running instead of being first class scheduled workloads that say, obey resource limits.
- There is also a wee bit of security improvement with a dedicated backend, the token given to run the workflows can be quite limited, vs. the "just give full access to docker.sock" approach of the default. I'm not implying this is a big deal, but it might be for anyone running "real" prod on Nomad.

In the end it's not a deal breaker for me, but I would start using it in an instant if it was available.
